sdk/turing/batch/config/source.py

```diff
@@ -0,0 +1,144 @@
+import abc
+from typing import Iterable, MutableMapping, Optional
+import turing.generated.models
+from turing.generated.model_utils import OpenApiModel
+from turing._base_types import DataObject
+
+
+class EnsemblingJobSource:
+    """
+    Configuration of source of the ensembling job
+    """
+
+    def __init__(self, dataset: 'Dataset', join_on: Iterable[str]):
+        self._dataset = dataset
+        self._join_on = join_on
+
+    @property
+    def dataset(self) -> 'Dataset':
+        return self._dataset
+
+    @property
+    def join_on(self) -> Iterable[str]:
+        return self._join_on
+
+    def to_open_api(self) -> OpenApiModel:
+        return turing.generated.models.EnsemblingJobSource(
+            dataset=self.dataset.to_open_api(),
+            join_on=self.join_on
+        )
+
+    def select(self, columns: Iterable[str]) -> 'EnsemblingJobPredictionSource':
+        """
+        Creates an instance of prediction source configuration
+
+        :param columns: list of columns from this source, that contain prediction data
+        :return: instance of `EnsemblingJobPredictionSource`
+        """
+        return EnsemblingJobPredictionSource(
+            self.dataset,
+            self.join_on,
+            columns
+        )
+
+
+class EnsemblingJobPredictionSource(EnsemblingJobSource):
+    """
+    Configuration of the prediction data for the ensembling job
+    """
+
+    def __init__(self, dataset, join_on, columns):
+        super(EnsemblingJobPredictionSource, self).__init__(dataset, join_on)
+        self._columns = columns
+
+    @property
+    def columns(self) -> Iterable[str]:
+        return self._columns
+
+    def to_open_api(self) -> OpenApiModel:
```

Comment:
In general, is it right to say that the contents in `sdk/turing/batch/...` act as a simple wrapper over the generated code, for the users to interact with, and the `to_open_api` and `from_open_api` allow for interfacing with the Turing API through the generated schema?

Reply:
Yes, correct. This wrapper is (hopefully) more user friendly and allows us to better control the user flow